src/plugins/intel_gna/tests/functional/shared_tests_instances/single_layer_tests/convolution.cpp

```diff
@@ -13,21 +13,115 @@ using namespace LayerTestsDefinitions;
 
 namespace {
 
-class gna_convolution_layerTest : public ConvolutionLayerTest {
+// ! [test_convolution:definition]
+typedef std::tuple<InferenceEngine::SizeVector,  // Kernel size
+                   InferenceEngine::SizeVector,  // Strides
+                   std::vector<ptrdiff_t>,       // Pad begin
+                   std::vector<ptrdiff_t>,       // Pad end
+                   InferenceEngine::SizeVector,  // Dilation
+                   size_t,                       // Num out channels
+                   ngraph::op::PadType           // Padding type
+                   >
+    convSpecificParams;
+
+typedef std::tuple<convSpecificParams,
+                   InferenceEngine::Precision,         // Net precision
+                   InferenceEngine::Precision,         // Input precision
+                   InferenceEngine::Precision,         // Output precision
+                   InferenceEngine::Layout,            // Input layout
+                   InferenceEngine::Layout,            // Output layout
+                   InferenceEngine::SizeVector,        // Input shapes
+                   LayerTestsUtils::TargetDevice,      // Device name
+                   std::map<std::string, std::string>  // GNA Config params
+                   >
+    ConvLayer2DTestParamsSet;
+
+class Convolution2DLayerTest : public testing::WithParamInterface<ConvLayer2DTestParamsSet>,
```

Comment:
I don't understand why we need to change the name of this class to 2D? It's generic enough to allow to us to test 1D convolutions as well. I strongly suggest not to change this. Can be changed in one of the next PRs.

Reply:
There is already ConvolutionLayerTest class and it is inherited by another test cases, because of that gtest complains that there is potential risk with the names.

Reply:
Clear, but then let's choose a name that doesn't cause conflicts and still doesn't create the false impression of being limited to 2D cases only.